hera_cal/datacontainer.py

```diff
@@ -84,12 +92,81 @@ def __delitem__(self, key):
         else:
             raise ValueError('only supports setting (ant1, ant2, pol) keys')
 
+    def __xor__(self, D):
+        """ concatenates DataContainers across the time [0] axis """
+        # check frequency structure matches
+        if D[D.keys()[0]].shape[1] != self.__getitem__(self.keys()[0]).shape[1]:
+            raise ValueError("[1] axis of dictionary values don't match")
+
+        # start new object
+        newD = odict()
+
+        # iterate over D keys
+        for i, k in enumerate(D.keys()):
+            if self.__contains__(k):
+                newD[k] = np.append(self.__getitem__(k), D[k], axis=0)
+
+        return DataContainer(newD)
   
-  
+    def __add__(self, D):
+        """ adds values of two DataContainers together """
+        # check time and frequency structure matches
+        if D[D.keys()[0]].shape[0] != self.__getitem__(self.keys()[0]).shape[0]:
+            raise ValueError("[0] axis of dictionary values don't match")
+        if D[D.keys()[0]].shape[1] != self.__getitem__(self.keys()[0]).shape[1]:
+            raise ValueError("[1] axis of dictionary values don't match")
+
+        # start new object
+        newD = odict()
+
+        # iterate over D keys
+        for i, k in enumerate(D.keys()):
+            if self.__contains__(k):
+                newD[k] = self.__getitem__(k) + D[k]
+
+        return DataContainer(newD)
+
+
+    def __pow__(self, D):
+        """ averages the values of two DataContainers together """
+        # check time and frequency structure matches
+        if D[D.keys()[0]].shape[0] != self.__getitem__(self.keys()[0]).shape[0]:
+            raise ValueError("[0] axis of dictionary values don't match")
+        if D[D.keys()[0]].shape[1] != self.__getitem__(self.keys()[0]).shape[1]:
+            raise ValueError("[1] axis of dictionary values don't match")
+
+        # start new object
+        newD = odict()
+
+        # iterate over D keys
+        for i, k in enumerate(D.keys()):
+            if self.__contains__(k):
+                newD[k] = (self.__getitem__(k) + D[k]) / 2.0
+
+        return DataContainer(newD)
+
+
+    def __mul__(self, D):
```

Comment:
maybe add a conjugate somewhere? either as an attribute to datacontainer itself or as an option. This is not necessary, but just suggestion.